src/DevAdventCalendarCompetition/DevAdventCalendarCompetition.TestResultService/TestResultRepository.cs

```diff
@@ -198,5 +198,34 @@ public ApplicationUser GetUserById(string id)
                 .Users
                 .FirstOrDefault(r => r.Id == id);
         }
+
+        /*  [in #1] ID of user that wants to know count of wrong answers
+         *  [in #2] ID of test that user wants to know count of wrong answers
+         * 
+         *  [out] Count of submitted wrong answers
+         */
+        public int GetWrongAnswersToTest(string userId, int testID) // get count of wrong answers
+        {
+            return _dbContext
+                .UserTestWrongAnswers
+                .Where(a => a.TestId == testID)
+                .Where(a => a.UserId == userId)
+                .Count();
+        }
+
+        /*  [in]#1 ID of user that wants to know when he submitted correct answer
+         *  [in]#2 ID of test that user wants to know when he submitted correct answer
+         * 
+         *  [out] Date of submission of corect answer
+         */
+        public DateTime GetCorrectAnswerDate(string userId, int testID) // get DateTime of correct answer
```

Comment:
I suggest renaming it to _GetUserTestCorrectAnswer_ - remember, that it can be null!!

Reply:
I'd say GetUserTestCorrectAnswer suggest that method returns whole CorrectAnswer instead of DateTime of CorrectAnswer. How about GetUserTestCorrectAnswerDate?